p2p/log.go

```diff
@@ -9,100 +14,131 @@ const (
 	MtDatasetLog = MsgType("dataset_log")
 )
 
+// DatasetLogRequestBody encapsulates options for requesting dataset history
+type DatasetLogRequestBody struct {
+	Ref    repo.DatasetRef
+	Limit  int
+	Offset int
+}
+
+// DatasetLogResponseBody encapsulates option for responding to a dataset history request
+type DatasetLogResponseBody struct {
+	History []repo.DatasetRef
+	Err     error
+}
+
 // RequestDatasetLog gets the log information of Peer's dataset
 func (n *QriNode) RequestDatasetLog(ref repo.DatasetRef, limit, offset int) ([]repo.DatasetRef, error) {
-	// id, err := n.Repo.Peers().IPFSPeerID(ref.Peername)
-	// if err != nil {
-	// 	return nil, fmt.Errorf("error getting peer IPFS id: %s", err.Error())
-	// }
-	// res, err := n.SendMessage(id, &Message{
-	// 	Type:    MtDatasetLog,
-	// 	Phase:   MpRequest,
-	// 	Payload: ref,
-	// })
-	// if err != nil {
-	// 	log.Debugf("send dataset log message error: %s", err.Error())
-	// 	return nil, err
-	// }
-
-	// data, err := json.Marshal(res.Payload)
-	// if err != nil {
-	// 	log.Debug(err.Error())
-	// 	return nil, err
-	// }
-
-	resref := []repo.DatasetRef{}
-	// err = json.Unmarshal(data, &resref)
-	// if len(resref) == 0 && err != nil {
-	// 	err = fmt.Errorf("no log found")
-	// }
-
-	return resref, nil
+
+	// get a list of peers to whom we will send the request
+	pids := n.ClosestConnectedPeers(ref.ProfileID, 15)
+	if len(pids) == 0 {
+		return nil, fmt.Errorf("no connected peers")
+	}
+
+	// create a channel on which to send the requests and receive the responses
```

Comment:
The comment on this line, and the 3 comments that follow don't really add anything, they're just restating what the code already says in a redundant manner. Please remove.